src/actions/MapActions.js

```diff
@@ -26,7 +27,7 @@ export const initializeMap = ({region}) => {
             .then(querySnapshot => {
                 const sites = _.map(querySnapshot.docs, (doc, index) => {
                     let preparedSite = _.clone(doc.data());
-                    preparedSite.id = index;
+                    preparedSite.id = doc.id;
```

Comment:
Using the real unique ID from Firebase for each site, instead of an arbitrary index